src/main/kotlin/three/consulting/epoc/service/FirebaseService.kt

```diff
@@ -55,15 +67,20 @@ class FirebaseService(
         }
     }
 
-    fun syncAllFirebaseUsers() {
+    fun syncAllFirebaseUsers(): MutableList<EmployeeDTO> {
 
         logger.info { "Starting to sync users found in Firebase with the database" }
 
+        val updatedUsers: MutableList<EmployeeDTO> = mutableListOf()
```

Comment:
Instead of using a mutable list you can return the `try` and use a map instead of the for loop. Also having two nested try blocks both catching generic exceptions seems very redundant. How about
```kotlin
fun syncAllFirebaseUsers(): MutableList<EmployeeDTO> {

    logger.info { "Starting to sync users found in Firebase with the database" }

    return try {
        val users = firebaseAuth.listUsers(null)
        users.map { user ->
            val syncedUser = syncFirebaseUser(user.uid, user.email, user.customClaims)
            logger.info { "synced user: $syncedUser" }
            syncedUser
        }.filterNotNull()
    } catch (e: Exception) {
        throw UnableToSyncFirebaseUsersException()
    }
}
```

Reply:
And you should log the exception before throwing the new one, so the stacktrace can be found on logs for debugging purposes
```kotlin
} catch (e: Exception) {
    logger.error(e) { "Failed to sync employees" }
    throw UnableToSyncFirebaseUsersException()
}
```